### PR TITLE
x264: rebuild with correct commit SHA

### LINKED ISSUE #64650: x264 wrong version
# Bug report

- [X] ran `brew update` and can still reproduce the problem?
- [X] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [X] ran `brew gist-logs <formula>`  ... no log
- [X] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` 

```
>brew config
HOMEBREW_VERSION: 2.5.9-52-g8cffae8
ORIGIN: https://github.com/Homebrew/brew
HEAD: 8cffae872ee0a29e2fc0a513c17410063194a18c
Last commit: 8 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 037d74722dc6038a8ac1f298cf966621b04d8e3e
Core tap last commit: 53 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.E8dHL6qEfG/org.macosforge.xquartz:0
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 12.0 build 1200
Git: 2.24.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
Java: 11.0.2
macOS: 10.15.7-x86_64
CLT: 12.2.0.0.1.1603499129
Xcode: N/A
XQuartz: 2.7.11 => /opt/X11
```
```
>brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed with `brew diy`.
You should find replacements for the following formulae:
  python@2

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libgps.dylib

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  python@2
  python@3.9

Warning: /usr/bin occurs before /usr/local/bin
This means that system-provided programs will be used instead of those
provided by Homebrew. Consider setting your PATH so that
/usr/local/bin occurs before /usr/bin. Here is a one-liner:
  echo 'export PATH="/usr/local/bin:$PATH"' >> ~/.zshrc

The following tools exist at both paths:
  smtpd2.7.py
  idle2.7
  phpize
  idle
  2to3-2.7
  python3
  phar.phar
  easy_install
  pydoc
  python
  python2
  pythonw
  pip3
  python2.7-config
  phar
  pydoc2.7
  pythonw2.7
  php
  python-config
  tidy
  easy_install-2.7
  python2.7
  smtpd.py
  php-config
```

## What you were trying to do (and why)

I need to install x264 r3027

## What happened (include command output)

x264 r2971M was installed. brew still thinks it's x264 r3027

<details>
  <summary>Command output</summary>
  <pre>

```
>x264 -h 
x264 core:157 r2971M 98ee9d2
Syntax: x264 [options] -o outfile infile
...
```

```
>brew list --versions x264
x264 r3027
```

  </pre>
</details>

## What you expected to happen

install x264 r3027 

## Step-by-step reproduction instructions (by running `brew install` commands)
```

>brew install x264
==> Downloading https://homebrew.bintray.com/bottles/x264-r3027.catalina.bottle.tar.gz
==> Downloading from https://d29vzk4ow07wi7.cloudfront.net/007c005fa1414651a0a0b2e753bee85ecbf0bc2242d7d56d0779a96712
######################################################################## 100.0%
==> Pouring x264-r3027.catalina.bottle.tar.gz
🍺  /usr/local/Cellar/x264/r3027: 11 files, 5.8MB
```

I tried brew cleanup and deleting the brew cache.

### PR DESCRIPTION
**Pull request by gromgit**
The original commit SHA (#64198) was for r2971. Also fixed license and removed patch that has been merged upstream (https://code.videolan.org/videolan/x264/-/merge_requests/35). Closes #64650.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by jonchang** [comment]
Let's add a test to make sure this doesn't happen again. Maybe...

`assert_match(version.lstrip("r"), shell_output("#{bin}/x264 --version").lines.first.chomp)`

(untested)?

File: Formula/x264.rb [hunk 1]
```diff
@@ -35,14 +36,6 @@ class X264 < Formula
...
-  # upstream PR https://code.videolan.org/videolan/x264/-/merge_requests/35
-  # Can be removed once it gets merged into stable branch
-  patch do
-    url "https://code.videolan.org/videolan/x264/-/commit/eb95c2965299ba5b8598e2388d71b02e23c9fba7.diff?full_index=1"
-    sha256 "7cdc60cffa8f3004837ba0c63c8422fbadaf96ccedb41e505607ead2691d49b9"
-  end
-
   def install
     # Work around Xcode 11 clang bug
     # https://bitbucket.org/multicoreware/x265/issues/514/wrong-code-generated-on-macos-1015
```
Comment (bayandin) [inline Formula/x264.rb:41]: I believe we don't need this workaround as well. It was fixed in Xcode

Comment (gromgit) [reply Formula/x264.rb]: OK, it seems to build and test OK with XCode 12.1 on my box. Thanks much!

**Comment by gromgit** [comment]
Both `libav` and `ffmpeg@2.8` have undeclared X11 linkage, and `--disable-libxcb` (the obvious configure option) doesn't seem to eliminate that linkage on my local box, so I'll add `libx11` instead to make things work.
```sh
$ brew linkage libav
...
Undeclared dependencies with linkage:
  libx11
Dependencies with no linkage:
  libvpx
  xvid

$ libav-12.3/configure --help | grep -i xcb
  --enable-libxcb            X window system protocol communication
  --enable-libxcb-shm        X11 shm communication [auto]
  --enable-libxcb-xfixes     X11 mouse rendering [auto]

$ brew linkage ffmpeg@2.8
...
Undeclared dependencies with linkage:
  libx11
Dependencies with no linkage:
  frei0r
  libvpx
  xvid

$ ffmpeg-2.8.17/configure --help | grep -i xcb
  --enable-libxcb          enable X11 grabbing using XCB [autodetect]
  --enable-libxcb-shm      enable X11 grabbing shm communication [autodetect]
  --enable-libxcb-xfixes   enable X11 grabbing mouse rendering [autodetect]
  --enable-libxcb-shape    enable X11 grabbing shape rendering [autodetect]
```

**Comment by fxcoudert** [comment]
> so I'll add libx11 instead to make things work

Let's not do that, we shouldn't pull X11 for these formulas. We try to use X11 as little as possible. There must be a way to deactivate X11 detection and linking.

**Comment by gromgit** [comment]
OK, everything checks out now, and X11 be gone. :sweat_smile:

**Review by SMillerDev — APPROVED** [review]

**Comment by fxcoudert** [comment]
This apparently didn't work: https://github.com/Homebrew/homebrew-core/pull/64941

```
==> /usr/local/Cellar/ffmpeg@2.8/2.8.17_3/bin/ffmpeg -y -filter_complex testsrc=rate=1:duration=1 /private/tmp/ffmpeg@2.8-test-20201116-33065-116ssds/video.mp4
dyld: Library not loaded: /usr/local/opt/libx11/lib/libX11.6.dylib
  Referenced from: /usr/local/Cellar/ffmpeg@2.8/2.8.17_3/bin/ffmpeg
  Reason: image not found
```

**Comment by gromgit** [comment]
Apologies, turns out `ffmpeg@2.8` didn't link with `libx11` on my local box because I'd unlinked `libx11` during a prior test. :facepalm:

With `libx11` linked, it looks like a couple of config flags are needed to explicitly turn off probing for X11 headers & libraries. Will push a PR to fix this permanently, once I triple-check it locally.

I'm curious though: How did `ffmpeg@2.8` find `libx11` during CI, when none of the formulae or their dependencies depend on it? It's not part of the standard CI environment, right?

**Comment by fxcoudert** [comment]
It's opportunistic linkage: when it's built as part of a multi-formula PR (for example a series of revision bumps because of a shared dependency) then all formulas are not necessarily removed

**Pull request closed without merging**